packages/@webex/plugin-cc/src/services/task/types.ts

```diff
@@ -0,0 +1,330 @@
+  TASK_UNHOLD: 'task:unhold',
+  TASK_CONSULT: 'task:consult',
+  TASK_CONSULT_END: 'task:consultEnd',
+  TASK_CONSULT_ACCEPT: 'task:consultAccepted',
+  TASK_PAUSE: 'task:pause',
+  TASK_RESUME: 'task:resume',
+  TASK_END: 'task:end',
+  TASK_WRAPUP: 'task:wrapup',
+} as const;
+
+export type TASK_EVENTS = Enum<typeof TASK_EVENTS>;
+
+type Interaction = {
+  isFcManaged: boolean;
+  isTerminated: boolean;
+  mediaType: MEDIA_CHANNEL;
+  previousVTeams: string[];
+  state: string;
+  currentVTeam: string;
+  participants: any; // todo
+  interactionId: string;
+  orgId: string;
+  createdTimestamp?: number;
+  isWrapUpAssist?: boolean;
+  callProcessingDetails: {
+    QMgrName: string;
+    taskToBeSelfServiced: string;
+    ani: string;
+    displayAni: string;
+    dnis: string;
+    tenantId: string;
+    QueueId: string;
+    vteamId: string;
+    pauseResumeEnabled?: string;
+    pauseDuration?: string;
+    isPaused?: string;
+    recordInProgress?: string;
+    recordingStarted?: string;
+    ctqInProgress?: string;
+    outdialTransferToQueueEnabled?: string;
+    convIvrTranscript?: string;
+    customerName: string;
+    virtualTeamName: string;
+    ronaTimeout: string;
+    category: string;
+    reason: string;
+    sourceNumber: string;
+    sourcePage: string;
+    appUser: string;
+    customerNumber: string;
+    reasonCode: string;
+    IvrPath: string;
+    pathId: string;
+    fromAddress: string;
+    parentInteractionId?: string;
+    childInteractionId?: string;
+    relationshipType?: string;
+    parent_ANI?: string;
+    parent_DNIS?: string;
+    consultDestinationAgentJoined?: boolean | string;
+    consultDestinationAgentName?: string;
+    parent_Agent_DN?: string;
+    parent_Agent_Name?: string;
+    parent_Agent_TeamName?: string;
+    isConferencing?: string;
+    monitorType?: string;
+    workflowName?: string;
+    workflowId?: string;
+    monitoringInvisibleMode?: string;
+    monitoringRequestId?: string;
+    participantInviteTimeout?: string;
+    mohFileName?: string;
+    CONTINUE_RECORDING_ON_TRANSFER?: string;
+    EP_ID?: string;
+    ROUTING_TYPE?: string;
+    fceRegisteredEvents?: string;
+    isParked?: string;
+    priority?: string;
+    routingStrategyId?: string;
+    monitoringState?: string;
+    BLIND_TRANSFER_IN_PROGRESS?: boolean;
+    fcDesktopView?: string;
+  };
+  mainInteractionId?: string;
+  media: Record<
+    string,
+    {
+      mediaResourceId: string;
+      mediaType: MEDIA_CHANNEL;
+      mediaMgr: string;
+      participants: string[];
+      mType: string;
+      isHold: boolean;
+      holdTimestamp: number | null;
+    }
+  >;
+  owner: string;
+  mediaChannel: MEDIA_CHANNEL;
+  contactDirection: {type: string};
+  outboundType?: string;
+  callFlowParams: Record<
+    string,
+    {
+      name: string;
+      qualifier: string;
+      description: string;
+      valueDataType: string;
+      value: string;
+    }
+  >;
+};
+
+export type TaskData = {
```

Comment:
Are we not storing the aux id here?

Reply:
It's present inside the agent profile, why do we need to keep it within the Task object